### PR TITLE
feat(pci-kubernetes): add modal for update/create/delete oidc provider

### PR DESCRIPTION
**Pull request by kqesar**
| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/pci-kubernetes`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | DTCORE-2452
| License          | BSD 3-Clause

- [ ] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [ ] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

## Description

feat(pci-kubernetes): add modal for update/create/delete oidc provider

### CODE REVIEW

File: packages/manager/apps/pci-kubernetes/src/api/hooks/useKubernetes.ts [hunk 1]
```diff
@@ -375,3 +379,103 @@ export const useResetCluster = ({
...
+  const mutation = useMutation({
+    mutationFn: async () => addOidcProvider(projectId, kubeId, params),
+    onError,
+    onSuccess: async () => {
+      await queryClient.invalidateQueries({
+        queryKey: getKubernetesClusterQuery(projectId, kubeId),
+      });
+      await queryClient.invalidateQueries({
+        queryKey: getAllKubeQueryKey(projectId),
+      });
```
Comment (frenautvh) [inline packages/manager/apps/pci-kubernetes/src/api/hooks/useKubernetes.ts:407]: does all kubes have to be invalidated ?

File: packages/manager/apps/pci-kubernetes/src/pages/oidc-provider/AddOIDCProvider.page.tsx [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+      ...formState,
+      clientIdError: formState.clientIdTouched && !formState.clientId,
+      issuerUrlError: formState.issuerUrlTouched && !formState.issuerUrl,
+    });
+  }, [
+    formState.clientId,
+    formState.clientIdTouched,
+    formState.issuerUrl,
+    formState.issuerUrlTouched,
+  ]);
```
Comment (frenautvh) [inline packages/manager/apps/pci-kubernetes/src/pages/oidc-provider/AddOIDCProvider.page.tsx:51]:
```suggestion
const hasClientIdError = formState.clientIdTouched && !formState.clientId;
const hasIssuerUrlError: formState.issuerUrlTouched && !formState.issuerUrl;
```
i don't think an useEffect and storing errors in formState is needed here as you can derive error state from the formState

File: packages/manager/apps/pci-kubernetes/src/pages/oidc-provider/UpdateOIDCProvider.page.tsx [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+  const [formState, setFormState] = useState({
+    issuerUrl: '',
+    issuerUrlError: false,
+    issuerUrlTouched: false,
+    clientId: '',
+    clientIdError: false,
+    clientIdTouched: false,
+  });
+  const { addError, addSuccess } = useNotifications();
+  useEffect(() => {
```
Comment (frenautvh) [inline packages/manager/apps/pci-kubernetes/src/pages/oidc-provider/UpdateOIDCProvider.page.tsx:43]: same here i don't think you need to store errors an use an useEffect

**Review by ghyenne — APPROVED** [review]

File: packages/manager/apps/pci-kubernetes/src/api/data/kubernetes.ts [hunk 4]
```diff
@@ -132,3 +132,34 @@ export const resetCluster = async (
   );
   return data;
 };
+
+export const addOidcProvider = async (
+  projectId: string,
+  kubeId: string,
+  params: Partial<TOidcProvider>,
```
Comment (seven-amid) [inline packages/manager/apps/pci-kubernetes/src/api/data/kubernetes.ts:139]: For a POST use, you need the whole object props, so you should remove the Partial

File: packages/manager/apps/pci-kubernetes/src/api/hooks/useKubernetes.ts [hunk 5]
```diff
@@ -375,3 +379,94 @@ export const useResetCluster = ({
     ...mutation,
   };
 };
+
+type AddOidcProviderProps = {
+  projectId: string;
+  kubeId: string;
+  params: Partial<TOidcProvider>;
```
Comment (seven-amid) [inline packages/manager/apps/pci-kubernetes/src/api/hooks/useKubernetes.ts:386]: No Partial here too, since it's an add operation

**Review by seven-amid — APPROVED** [review]

**Pull request merged**